src/console/commands/migration.commands.ts

```diff
@@ -0,0 +1,81 @@
+import {InjectQueue} from "@nestjs/bull";
+
+import {Command, Console} from "nestjs-console";
+import {LumTypes} from "@lum-network/sdk-javascript";
+import {Queue} from "bull";
+
+import {BlockService, ElasticsearchService, LumNetworkService, ValidatorService} from "@app/services";
+import {QueueJobs, Queues} from "@app/utils";
+
+
+@Console({command: 'migration', description: 'Migration related commands'})
+export class MigrationCommands {
+    constructor(
+        @InjectQueue(Queues.QUEUE_BLOCKS) private readonly _queue: Queue,
+        private readonly _blockService: BlockService,
+        private readonly _chainService: LumNetworkService,
+        private readonly _elasticSearch: ElasticsearchService,
+        private readonly _validatorService: ValidatorService
+    ) {
+
+    }
+
+    async* scrollSearch(params) {
+        let response = await this._elasticSearch.client.search(params);
+        while (true) {
+            const sourceHits = response.body.hits.hits
+
+            if (sourceHits.length === 0) {
+                break
+            }
+
+            for (const hit of sourceHits) {
+                yield hit
+            }
+
+            if (!response.body._scroll_id) {
+                break
+            }
+
+            response = await this._elasticSearch.client.scroll({
+                scroll_id: response.body._scroll_id,
+                scroll: params.scroll
+            });
+        }
+    }
+
+    @Command({command: 'blocks', description: 'Migrate blocks from Elasticsearch to Postgres'})
+    async migrateBlocks(): Promise<void> {
+        const chainId = await this._chainService.client.getChainId();
+
+        console.log('starting');
```

Comment:
Don't we prefer a logger function here?

Reply:
The Logger interface inherited form NestJS is not available in the Console context...